R/get_code.R

```diff
@@ -92,6 +93,38 @@ get_code.TealDataAbstract <- function(x, dataname = character(0), deparse = TRUE
   }
 }
 
+#' @rdname get_code
+#' @export
+#' @examples
+#'
+#' tdata1 <- teal_data()
+#' tdata1 <- within(tdata1, {
+#'   a <- 1
+#'   b <- a^5
+#'   c <- list(x = 2)
+#' })
+#' get_code(tdata1)
+#' get_code(tdata1, "a")
+#' get_code(tdata1, "b")
+#'
+get_code.teal_data <- function(x, dataname = character(0), deparse = TRUE, ...) {
+  if (!x@valid) {
+    warning("Code was not validated for reproducibility.")
```

Comment:
This is not up to specs.

> 4\. `get_code` method should add extra message on top `warning("code is not valid")` - please rephrase

Reply:
Or either the specs are not clear. You can understand this as
- a) print extra message to console when `get_code` is run and `valid` = FALSE
- b) pre-pend the returned `code` with `warning()` statement

If this is b) then I can pre-pend `code` with  `code <-  c('warning("code is not valid")', x@code)`

Reply:
It clearly means the latter.

Reply:
Changed the way you proposed, and it looks like it has more sense to return this warning with the code and not in the console 🗡️ 

```r
> tdata2 <- teal_data(x1 = iris, code = "x1 = iris")
> get_code(tdata2)
[1] "warning('Code was not validated for reproducibility.')"
[2] "x1 = iris"                                             
> get_code(is_reproducible(tdata2))
[1] "x1 = iris"
```

08ec33e97beb704b632eb3dd85acda61dc17ac1e